### PR TITLE
8270317: Large Allocation in CipherSuite

### PR DESCRIPTION
**Pull request by phohensee**
Backport for Oracle parity. Backport from 11u is not clean due to lack of CopyOf in 8u. The nearest equivalents are Collections.unmodifiableMap and Collections.unmodifiableList. Also, there is no jmh archive in 8u, so the performance test is not included.

Tier1 and all security tests pass, in production at Amazon for over a year without issues.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8270317](https://bugs.openjdk.org/browse/JDK-8270317): Large Allocation in CipherSuite

### Reviewers
 * [Volker Simonis](https://openjdk.org/census#simonis) (@simonis - **Reviewer**) ⚠️ Review applies to [c2debd86](https://git.openjdk.org/jdk8u-dev/pull/253/files/c2debd8635b676378b546e885c0d5d348a2baf65)
 * [Martin Balao](https://openjdk.org/census#mbalao) (@martinuy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/253/head:pull/253` \
`$ git checkout pull/253`

Update a local copy of the PR: \
`$ git checkout pull/253` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/253/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 253`

View PR using the GUI difftool: \
`$ git pr show -t 253`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/253.diff">https://git.openjdk.org/jdk8u-dev/pull/253.diff</a>

</details>

### CODE REVIEW

File: jdk/src/share/classes/sun/security/ssl/CipherSuite.java
Comment (simonis) [inline jdk/src/share/classes/sun/security/ssl/CipherSuite.java:904]: Agree that in this case `Collections.unmodifiableMap()`/`Collections.unmodifiableList()` are a good replacement for the missing `copyOf()`. And it should also be faster, because it avoids a copy :)

**Review by simonis — APPROVED** [review]
Looks good except for the comment change (which might be a merge artifact?).

**Review by martinuy — APPROVED** [review]
Hi Paul,

Thanks for your contribution. Looks good to me too.

Martin.-

**Comment by phohensee** [comment]
Thank you, Martin. Now just waiting on maintainer review/approval...

**Comment by phohensee** [comment]
/integrate

Got the approval.

**Pull request closed without merging**
